dvci/shc-web-verifier
# Dvci 170 - Lab Result Verification

## Overview
This PR allows our app to scan and display SHCs that contain Lab Results. Credentials of [type #laboratory](https://spec.smarthealth.cards/vocabulary/) that comply with the [Generic Laboratory Result Observation profile](https://build.fhir.org/ig/HL7/fhir-shc-vaccination-ig/StructureDefinition-shc-infectious-disease-laboratory-result-observation-dm.html), can now be scanned, verified and displayed back to the user.

## What Changed
- Added `LabCard` component that gets used in `HealthCardDisplay` when a VC of type https://smarthealth.cards#laboratory is scanned
- `LabCard` component displays the `code`, `effectiveDateTime`, `value[x]`, `referenceRange`, and `performer` of the Observations in the SHC. Helper functions were added to handle elements that could have multiple data types. `code` and `value[x]` draw display values from the value sets that live in `src/components/LabCard`.
- An error is thrown if both #laboratory and #immunization types are found in the VC's `type` array, as we decided it is against the Vaccination and Testing IG to allow both in one Bundle.
- `LabCard.test.js` was added to test UI rendering of the LabCard component.

## To Test
- Make sure lab data is displaying appropriately
   -  Here is a test lab QR code that has all Must Support fields populated:
   - ![image](https://user-images.githubusercontent.com/51176978/180104569-4ba2143f-9ebf-4ece-a02f-1f03aeb071ff.png)
   - See [DVCI-170](https://jira.mitre.org/browse/DVCI-170) for a link to a real SHC.
- Check that the fields being displayed by the `LabCard` correspond to what would be expected from a resource conforming to the [Generic Lab Results profile](https://build.fhir.org/ig/HL7/fhir-shc-vaccination-ig/StructureDefinition-shc-infectious-disease-laboratory-result-observation-dm.html).
- Test different edge cases and error scenarios using our [Lab Result QR Code examples ](https://confluence.mitre.org/display/DVCI/Lab+Result+QR+Codes+for+Testing+the+Web+Verifier) on Confluence.
- Make sure you are still able to scan vaccine cards, and that no functionality was broken.
